### PR TITLE
add required option

### PR DESCRIPTION
**Pull request by SandraRoDiaz**

### CODE REVIEW

File: src/autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -120,6 +120,13 @@ type autocompleteProps = {
    * onRemoveAll of the selected listItems
    */
   onRemoveAll?: () => void;
+
+  /**
+   * Optional required flag
+   */
+
```
Comment (daniele-zurico) [inline src/autocomplete/Autocomplete.tsx:127]: can you please remove this empty line?

File: src/autocomplete/__tests__/Autocomplete.test.tsx [hunk 2]
```diff
@@ -353,4 +353,26 @@ describe('FormInput', () => {
 
     expect(onRemoveAllHandler).toHaveBeenCalled();
   });
+
+  it('should check that required attribute is defaulted to false', () => {
+
```
Comment (daniele-zurico) [inline src/autocomplete/__tests__/Autocomplete.test.tsx:358]: same here can we remove all the empty lines?

File: src/formInput/FormInput.tsx [hunk 3]
```diff
@@ -103,6 +103,12 @@ type FormInputProps = {
    * allow to define an hint
    */
   hint?: HintProps;
+
+   /**
+   * Optional required flag
+   */
+
```
Comment (daniele-zurico) [inline src/formInput/FormInput.tsx:110]: same here can we remove all the empty lines?

File: src/autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -120,6 +120,10 @@ type autocompleteProps = {
    * onRemoveAll of the selected listItems
    */
   onRemoveAll?: () => void;
+  /**
+   * Optional required flag
```
Comment (Ibabalola) [inline src/autocomplete/Autocomplete.tsx:124]: I feel we need a more descriptive comment here, required on its own doesn't reveal it's intended purpose

File: src/formInput/FormInput.tsx [hunk 5]
```diff
@@ -103,6 +103,10 @@ type FormInputProps = {
    * allow to define an hint
    */
   hint?: HintProps;
+   /**
+   * Optional required flag
```
Comment (Ibabalola) [inline src/formInput/FormInput.tsx:107]: same - I feel we need a more descriptive comment here, required on its own doesn't reveal it's intended purpose

File: stories/Autocomplete/Styled.stories.mdx [hunk 6]
```diff
@@ -159,6 +159,47 @@ The list of available options are:
...
+          hintText="Select your fruit"
+          hintClass="autocomplete__label"
+          inputProps={{ className: 'govuk-input' }}
+          resultUlClass="autocomplete__menu"
+          resultlLiClass="autocomplete__option"
+          resultNoOptionClass="resultNoOptionClass"
+          resultActiveClass="autocomplete__option"
+          notFoundText="No fruit found"
+          onSelected={handleSelected}
+          required
```
Comment (Ibabalola) [inline stories/Autocomplete/Styled.stories.mdx:194]: please update the Documentation.mdx also for Autocomplete and FormInput with the new required field.

**Review by Ibabalola — CHANGES_REQUESTED** [review]

File: src/autocomplete/Autocomplete.tsx (hunk 4 above)
Comment (daniele-zurico) [reply src/autocomplete/Autocomplete.tsx:124]: maybe something like:
`specifies if that an input field need to be filled ` could probably do the job?

**Review by Ibabalola — APPROVED** [review]

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**